common/web/gesture-recognizer/src/touchEventEngine.ts

```diff
@@ -68,37 +68,76 @@ namespace com.keyman.osk {
       }
     }
 
+    public cleanupSequenceWithId(identifier: number) {
+      super.cleanupSequenceWithId(identifier);
+
+      delete this.safeBoundMaskMap[identifier];
+    }
+
+    private buildSampleFromTouch(touch: Touch): InputSample {
+      return this.buildSampleFor(touch.clientX, touch.clientY);
+    }
+
     onTouchStart(event: TouchEvent) {
-      this.preventPropagation(event);
-      const coord = InputEventCoordinate.fromEvent(event)
-
-      if(!ZoneBoundaryChecker.inputStartOutOfBoundsCheck(coord, this.config)) {
-        // If we started very close to a safe zone border, remember which one(s).
-        // This is important for input-sequence cancellation check logic.
-        this.disabledSafeBounds = ZoneBoundaryChecker.inputStartSafeBoundProximityCheck(coord, this.config);
-      } else {
-        this.disabledSafeBounds = 0;
-        // TODO:  disable tracking for the specific `Touch` that just started.
-        // Requires explicit implementation of multi-touch tracking, which has not yet been added.
+      // If it's not an event we'd consider handling, do not prevent event
+      // propagation!  Just don't process it.
+      if(!this.config.targetRoot.contains(event.target as Node)) {
+        return;
       }
 
-      this.onInputStart(coord);
+      this.preventPropagation(event);
+
+      for(let i=0; i < event.changedTouches.length; i++) {
```

Comment:
It's not possible to do `for(const touch of event.changedTouches) {`? Ah, I see https://github.com/microsoft/TypeScript/issues/31826

Reply:
Yep.  I find myself wondering about that DOM design decision.